[CI] InboundHandlerTests testLogsSlowInboundProcessing failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6205 / ubuntu-2404-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/u3ry4lxw2nvkw)
- [elasticsearch-periodic-platform-support #6205 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/34x2siw36blq2)
- [elasticsearch-periodic-platform-support #6205 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/mareempcpyjbk)
- [elasticsearch-periodic-platform-support #6205 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/lzroyraaqwqoe)
- [elasticsearch-periodic-platform-support #6205 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/qvcozdp7ll2oy)
- [elasticsearch-periodic #6216 / release-tests](https://gradle-enterprise.elastic.co/s/lrvfc5d2aom24)
- [elasticsearch-periodic #6216 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/7j5zfyhej32yq)
- [elasticsearch-periodic #6216 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/6aeqnw6bxqgci)
- [elasticsearch-periodic #6216 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/2eegs2lptornq)
- [elasticsearch-periodic-platform-support #6205 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/7uwbozoztwtko)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.transport.InboundHandlerTests.testLogsSlowInboundProcessing" -Dtests.seed=33B577DA79AC6216 -Dtests.locale=el-Grek-GR -Dtests.timezone=Asia/Nicosia -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.transport.InboundHandlerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testLogsSlowInboundProcessing),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected to see expected slow request but did not
Expected: <0L>
     but: was <1L>
```

**Issue Reasons:**
- [main] 6 consecutive failures in test testLogsSlowInboundProcessing
- [main] 30 failures in test testLogsSlowInboundProcessing (13.3% fail rate in 225 executions)
- [main] 5 failures in step part-1 (5.7% fail rate in 88 executions)
- [main] 2 failures in step part1 (6.9% fail rate in 29 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (5.8% fail rate in 86 executions)
- [main] 2 failures in pipeline elasticsearch-intake (6.9% fail rate in 29 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -387,9 +387,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/121672
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   issue: https://github.com/elastic/elasticsearch/issues/121411
-- class: org.elasticsearch.transport.InboundHandlerTests
-  method: testLogsSlowInboundProcessing
-  issue: https://github.com/elastic/elasticsearch/issues/121816
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/cat/health/cat-health-no-timestamp-example}
   issue: https://github.com/elastic/elasticsearch/issues/121867

--- a/server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java
@@ -291,8 +291,11 @@ public class InboundHandlerTests extends ESTestCase {
             BytesStreamOutput byteData = new BytesStreamOutput();
             TaskId.EMPTY_TASK_ID.writeTo(byteData);
             // simulate bytes of a transport handshake: vInt transport version then release version string
-            TransportVersion.writeVersion(remoteVersion, byteData);
-            byteData.writeString(randomIdentifier());
+            try (var payloadByteData = new BytesStreamOutput()) {
+                TransportVersion.writeVersion(remoteVersion, payloadByteData);
+                payloadByteData.writeString(randomIdentifier());
+                byteData.writeBytesReference(payloadByteData.bytes());
+            }
             final InboundMessage requestMessage = new InboundMessage(
                 requestHeader,
                 ReleasableBytesReference.wrap(byteData.bytes()),

